static_report/src/components/ComparisonReport/CRTabProfilingDetails.tsx

```diff
@@ -54,53 +55,62 @@ type CRProfilingColumnProps = {
   target?: ColumnSchema;
 };
 function CRProfilingColumn({ name, base, target }: CRProfilingColumnProps) {
-  const [data, setData] = useState<(CRHistogramDatum[] | null)[]>([]);
+  const [combinedData, setCombinedData] = useState<CRHistogramDatum[] | null>(
+    null,
+  );
+  const [splitData, setSplitData] = useState<
+    (BarChartDatum[] | null | undefined)[]
+  >([]);
 
   useEffect(() => {
-    if (
-      base?.type === target?.type &&
-      (base?.type === 'string' || base?.type === 'datetime')
-    ) {
+    const isSameGenericType = base?.type === target?.type;
+    const isStringOrDatetime =
+      base?.type === 'string' || base?.type === 'datetime';
+
+    // Determine combined data histograms
+    if (isSameGenericType && isStringOrDatetime) {
       const transformResult = transformCRStringDateHistograms({
         base: base?.histogram,
         target: target?.histogram,
       });
 
-      setData([transformResult]);
+      setCombinedData(transformResult);
     } else {
-      const baseData = base
-        ? transformBaseHistogram({
-            baseCounts: base?.histogram?.counts,
-            baseLabels: base?.histogram?.labels,
-          })
-        : null;
-
-      const targetData = target
-        ? transformBaseHistogram({
-            baseCounts: target?.histogram?.counts,
-            baseLabels: target?.histogram?.labels,
-          })
-        : null;
+      // Determine split data histograms
+      const transformBaseResult = getColumnTypeChartData(base);
+      const transformTargetResult = getColumnTypeChartData(target);
 
-      setData([baseData, targetData]);
+      // Needs to show mismatched columns (null | undefined)
+      setSplitData([transformBaseResult, transformTargetResult]);
     }
   }, [base, target]);
 
-  const isSingleChartOnly = data.length === 1;
+  // Show combined base|target chart or split charts
   return (
     <Flex key={name} direction="column">
       <Grid my={8} templateColumns="500px 1fr" gap={12}>
         <CRTableColumnDetails baseColumn={base} targetColumn={target} />
 
         <Grid
           my={4}
-          templateColumns={`1fr ${isSingleChartOnly ? '' : '1fr'}`}
-          gap={isSingleChartOnly ? 0 : 12}
+          templateColumns={`1fr ${combinedData ? '' : '1fr'}`}
+          gap={combinedData ? 0 : 12}
+          className="foobarbaz"
```

Comment:
Is this `className` for testing purposes XD?

Reply:
exactly ;)